File: Level 1/기사단원의 무기/solution.js
function solution(number, limit, power) {
  const numbers = Array.from({ length: number }, (_, i) => i + 1);
  const measureCounts = numbers.map(getMeasureCount);
  return measureCounts.reduce(
    (acc, cur) => acc + (cur > limit ? power : cur),
    0
  );
}

function getMeasureCount(number) {
  let measureCount = 0;

  for (let i = 1; i <= Math.sqrt(number); i++) {
    if (number % i === 0) {
      measureCount++;

      if (i * i !== number) {
        measureCount++;
      }
    }
  }

  return measureCount;
}

File: Level 1/기사단원의 무기/solution.py
import math


def get_measure_count(number):
    count = 0

    for i in range(1, int(math.sqrt(number)) + 1):
        if number % i == 0:
            count += 1
            if i * i != number:
                count += 1

    return count


def solution(number, limit, power):
    measure_counts = [get_measure_count(i) for i in range(1, number + 1)]
    return sum(power if count > limit else count for count in measure_counts)
